lib/api.js

```diff
@@ -115,9 +130,20 @@ function makeApiRequest(transport, opts, bodyObj, callback) {
     });
   });
 
+  var originalArguments = arguments;
   req.on('error', function (err) {
-    logger.error('Could not make request to rollbar, ' + err);
-    callback(err);
+    // If the request to Rollbar failed due to a connection error, lets queue
+    // up the requests and try again periodically in the hopes that the connection
+    // will be restored.
+    //
+    // If no retryInterval set, interpret that as not wanting to retry send errors,
+    // so just bail out silently.
+    if (SETTINGS.retryInterval && err.code == 'ENOTFOUND') {
```

Comment:
I think there are some we want to fail on, I looked around and found
```
var RETRIABLE_ERRORS = ['ECONNRESET', 'ENOTFOUND', 'ESOCKETTIMEDOUT', 'ETIMEDOUT', 'ECONNREFUSED', 'EHOSTUNREACH', 'EPIPE', 'EAI_AGAIN'];
```
As a list of network errors from https://github.com/FGRibreau/node-request-retry/blob/master/strategies/NetworkError.js

Let's just use something like
```
var shouldRetry = false;
if (SETTINGS.retryInterval) {
  for (var i=0, j=RETRIABLE_ERRORS.length; i < j; i++) {
    if (err.code === RETRIABLE_ERRORS[i]) {
      shouldRetry = true;
      break;
    }
  }
}
if (shouldRetry) {
  pushOntoRetryQueue.apply(null, originalArguments);
} else {
  logger.error('Could not make request to rollbar, ' + err);
  callback(err);
}
```

I am not using `Array.prototype.indexOf` because apparently it is only IE9+ and I just fixed something else that was making us not work in IE8.